### ISSUE TITLE
Sign-In with Ethereum Support

### ISSUE THREAD

**Issue by expede** [ucan-wg/spec#11]
Just a stub for now: UCAN support / flows for [Sign in With Ethereum](https://github.com/ethereum/EIPs/blob/9a9c5d0abdaf5ce5c5dd6dc88c6d8db1b130e95b/EIPS/eip-4361.md)

**Comment by oed** [comment]
Would be interested in collaborating here as we have created [CACAO](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-draft_cacao.md) as a way to use SIWE for OCAPs.

cc @ukstv @clehner

**Comment by expede** [comment]
Awesome; _I'd love to hear_ how you've all been thinking about it!

It's not written down here, but our rough idea is to use a deterministic template, fill that in and sign it. We can then embed the resulting signature in the facts field if we need to pass that challenge around.

There are a few use cases for this, which is the main area where we need to figure out what exactly to support: proving the signature for a UCAN plugging into existing SIWE tools, tokens to hydrate the template from, passing around challenges, and extending claims about access to wallets to other keys in the PK graph.

---

@oed

On another note, we may have some interesting ideas to share with y'all about making tokens DHT-, transport- and blockstore-agnostic, more compact (with a Merkle compression protocol), and handing wall clock time bounds in eventually consistent systems like Blockchains and CRDTs.

In short, yes I'd love to sync up sometime :)

**Comment by expede** [comment]
Gave that a read. haha so pretty much the same strategy in CACAO — amazing! Clearly we're on to something 😉

**Comment by expede** [comment]
Closing in favour of exploring UCAN<>CACAO integration post-CASA

**Comment by dholms** [comment]
Is there somewhere to follow along on that convo 👀

**Comment by dholms** [comment]
Ah wait think I found it: https://github.com/ipld/js-dag-ucan/issues/24

**Comment by oed** [comment]
The link above is mostly about using CACAO as a byte representation in IPLD. We also talked about how you could include a CACAO/SIWE message as a proof in an attenuation.
Here is some sparse notes: https://github.com/ChainAgnostic/AMS-CASA-gathering/issues/19
